tee-worker/service/src/prometheus_metrics.rs

```diff
@@ -182,7 +196,57 @@ impl ReceiveEnclaveMetrics for EnclaveMetricsReceiver {
 	}
 }
 
-// Data structure that matches with REST API JSON
+// Function to increment STF calls with labels
+fn inc_stf_calls(category: &str, label: &str) {
+	ENCLAVE_STF_CALLS.with_label_values(&[category, label]).inc();
+}
+
+// Function to observe STF call execution time with labels
+fn observe_execution_time(category: &str, label: &str, time: f64) {
+	ENCLAVE_STF_CALLS_EXECUTION.with_label_values(&[category, label]).observe(time);
+}
+
+// Handle STF call request and increment metrics
+fn handle_stf_call_request(req: RequestType, time: f64) {
+	// Determine the category based on the request type
+	let category = match req {
+		RequestType::IdentityVerification(_) => "link_identity",
+		RequestType::AssertionVerification(_) => "request_vc",
+	};
+
+	match req {
```

Comment:
This `match` can return `label` and this would lead to less duplication


```
fn handle_stf_call_request(req: RequestType, time: f64) {
	// Determine the category based on the request type
	let category = match req {
		RequestType::IdentityVerification(_) => "link_identity",
		RequestType::AssertionVerification(_) => "request_vc",
	};
	// Determine label based on request type and identity
	let label = match req {
		RequestType::IdentityVerification(request) => {
			match request.identity {
				Identity::Twitter(_) => "Twitter",
				Identity::Discord(_) => "Discord",
				Identity::Github(_) => "Github",
				Identity::Substrate(_) => "Substrate",
				Identity::Evm(_) => "Evm",
			}
		},
		RequestType::AssertionVerification(request) => {
			match request.assertion {
				Assertion::A1 => "A1",
				Assertion::A2(_) => "A2",
				Assertion::A3(..) => "A3",
				Assertion::A4(_) => "A4",
				Assertion::A6 => "A6",
				Assertion::A7(_) => "A7",
				Assertion::A8(_) => "A8",
				Assertion::A9 => "A9",
				Assertion::A10(_) => "A10",
				Assertion::A11(_) => "A11",
				Assertion::A13(_) => "A13",
				Assertion::A14 => "A14",
				Assertion::Achainable(..) => "Achainable",
			}
		},
	};
	inc_stf_calls(category, label);
	observe_execution_time(category, label, time)
}
```